src/realm/util/file.cpp

```diff
@@ -128,7 +128,22 @@ struct WindowsFileHandleHolder {
 #endif
 
 #if REALM_HAVE_STD_FILESYSTEM
+#if __cplusplus < 202002L
 using std::filesystem::u8path;
+inline std::string path_to_string(const std::filesystem::path& path)
+{
+    return path.u8string();
+}
+#else
+inline std::string path_to_string(const std::filesystem::path& path)
+{
+    return reinterpret_cast<const char*>(path.u8string().c_str());
+}
+inline auto u8path(const std::string& str)
+{
+    return std::filesystem::path(reinterpret_cast<const char8_t*>(str.c_str()));
```

Comment:
I was a bit concerned that this converts using the constructor that terminates at the first null byte, regardless of the length of the original string. But I suppose this is not actually a problem because null bytes cannot be in a file path.